src/server/status/server_status.js

```diff
@@ -2,15 +2,24 @@ import _ from 'lodash';
 
 import states from './states';
 import Status from './status';
+import PluginStatus from './plugin_status';
 
 module.exports = class ServerStatus {
   constructor(server) {
     this.server = server;
     this._created = {};
   }
 
-  create(plugin) {
-    return (this._created[plugin.id] = new Status(plugin, this.server));
+  create(id) {
```

Comment:
I like this! It more consistently mirrors the class hierarcy where everything is a `Status`, just that if a status has more specific properties like in the case of plugins, there's a subclass for it.